PurrSoft/PurrSoft.Api/Controllers/AnimalFosterMapController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PurrSoft.Api.Controllers.Base;
+using PurrSoft.Application.Common;
+using PurrSoft.Application.Models;
+using PurrSoft.Application.Queries.AnimalFosterMapQueries;
+using PurrSoft.Application.QueryOverviews;
+using System.Net;
+using static PurrSoft.Application.Commands.AnimalFosterMapCommands.AnimalFosterMapCommands;
+
+namespace PurrSoft.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class AnimalFosterMapController : BaseController
+{
+	[HttpGet()]
+	[Authorize(AuthenticationSchemes = "Bearer", Roles = "Manager, Foster")]
+	[ProducesResponseType(typeof(CollectionResponse<AnimalFosterMapDto>), (int)HttpStatusCode.OK)]
+	public async Task<CollectionResponse<AnimalFosterMapOverview>> GetAnimalFosterMaps([FromQuery] GetFilteredAnimalFosterMapsQueries query)
+	{
+		return await Mediator.Send(query, new CancellationToken());
+	}
+
+	[HttpGet("{id}")]
+	[Authorize(AuthenticationSchemes = "Bearer", Roles = "Manager, Foster")]
+	[ProducesResponseType(typeof(AnimalFosterMapDto), (int)HttpStatusCode.OK)]
+	[ProducesResponseType((int)HttpStatusCode.BadRequest)]
```

Comment:
You could say that the type of the response is a CommandResponse

Reply:
Isn't it fine for a typeof(AnimalFosterMapDto) response if that is the entity returned with the Ok Status code?

Reply:
i was referring to the BadRequest response, sorry for not specifying